[linux] Fix jsoncpp deprecations

vially opened the pull request:
The `jsoncpp` [Reader](http://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_reader.html) and [FastWriter](http://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_fast_writer.html) classes are [depreacted](http://open-source-parsers.github.io/jsoncpp-docs/doxygen/deprecated.html).

Because of this `flutter-desktop-embedding` fails to compile using `jsoncpp`-`v1.8.4` on ArchLinux.

This PR attempts to fix this by using the newer `jsoncpp` API.

PS: Please review this carefully because I don't know `C` and I'm unable to run this code in order to test it. It does seem to compile though.

stuartmorgan commented on linux/library/src/flutter_embedder.cc:
```diff
@@ -81,13 +81,15 @@ static void GLFWOnFlutterPlatformMessage(const FlutterPlatformMessage *message,
   std::string raw_string(reinterpret_cast<const char *>(message->message), 0,
                          message->message_size);
-  bool parsing_successful = parser.parse(raw_string, json);
+  bool parsing_successful = parser->parse(raw_string.c_str(), raw_string.c_str() + raw_string.size(), &json, &parse_errors);
```
This and 92 will need to be wrapped to follow the C++ style guide length limits.

https://google.github.io/styleguide/cppguide.html

But it also doesn't seem like there's anything to be gained by round-tripping through std::string with this API.

stuartmorgan commented on linux/library/src/flutter_embedder.cc:
```diff
@@ -99,8 +101,8 @@ static void GLFWOnFlutterPlatformMessage(const FlutterPlatformMessage *message,
 
-  Json::FastWriter writer;
-  std::string output = writer.write(response);
+  Json::StreamWriterBuilder wbuilder;
```
'wbuilder' (and 'rbuilder' above) will similarly need to be changed to meet the style guide guidance on not abbreviating names.

stuartmorgan commented on linux/library/src/flutter_embedder.cc:
```diff
@@ -99,8 +101,8 @@ static void GLFWOnFlutterPlatformMessage(const FlutterPlatformMessage *message,
-  Json::FastWriter writer;
-  std::string output = writer.write(response);
+  Json::StreamWriterBuilder wbuilder;
+  std::string output = Json::writeString(wbuilder, response);
```
According to the docs this is creating and destroying a new writer under the hood, much like the code above. If we have to heap-allocate reader/writer objects anyway, maybe we should just cache them? Unless the embedding API sends this on arbitrary threads (which I don't think is the case) it would be pretty easy.

awdavies replied:
At the moment everything should be running in the same thread, so it should be fine.

awdavies commented:
I implemented things this way just from checking out the code from github and building the docs.  The deprecated link didn't mention anything about it, but double-checking the headers it does indicate to use StreamWriterBuilder instead :+1:

vially commented:
@awdavies @stuartmorgan I've addressed most of your comments except caching the reader/writer. I can take a stab at that as well if you feel it's important.

awdavies commented:
Verified it compiles and runs :+1:

awdavies commented on linux/library/src/flutter_embedder.cc:
```diff
@@ -81,13 +81,16 @@ static void GLFWOnFlutterPlatformMessage(const FlutterPlatformMessage *message,
   GLFWwindow *window = reinterpret_cast<GLFWwindow *>(user_data);
-  Json::Reader parser;
+  Json::CharReaderBuilder reader_builder;
+  std::unique_ptr<Json::CharReader> const parser(reader_builder.newCharReader());
```
Remove the `const` here:

```c++
std::unique_ptr<Json::CharReader> parser(reader_builder.newCharReader());
``` 

Should be fine.

vially replied:
Fixed
